disco/models/upgrade_cost_analysis_generic_input_model.py

```diff
@@ -485,6 +485,75 @@ def check_timepoint_multipliers(cls, timepoint_multipliers):
         return timepoint_multipliers
 
 
+class PyDssControllerModels(UpgradeParamsBaseModel):
+    """Defines the settings for PyDSS controllers"""
+
+    pv_controller: Optional[PvControllerModel] = Field(
+        title="pv_controller", description="Settings for a PV controller"
+    )
+
+
+class UpgradeSimulationParamsModel(UpgradeParamsBaseModel):
+    """Parameters for all jobs in a simulation"""
+    plot_violations: bool = Field(
+        title="plot_violations",
+        description="If True, create plots of violations before and after simulation.",
+        default=True,
+    )
+    upgrade_order: Optional[List[str]] = Field(
+        description="Order of upgrade algorithm. 'thermal' or 'voltage' can be removed from the "
+        "simulation by excluding them from this parameter.",
+        default=_SUPPORTED_UPGRADE_TYPES,
+    )
+    include_pf1: Optional[bool] = Field(
+        title="include_pf1",
+        description="Include PF1 scenario (no controls) if pydss_controllers are defined.",
+        default=True,
+    )
+    dc_ac_ratio: Optional[float] = Field(
+        title="dc_ac_ratio", 
+        description="Apply DC-AC ratio for PV Systems", 
+        default=None
+    )
+    timeseries_analysis: Optional[bool] = Field(
+        title="timeseries_analysis", 
+        description="timeseries_analysis", 
+        default=False
+    )
+    timeseries_metadata: Optional[str] = Field(
```

Comment:
Yes, it is the metadata file. 
My idea is that this data is available in case needed, and can be processed alongwith the other outputs in the output.json file. It will mainly be on the user's discretion on how they want to visualize this later. It can be used in different ways and will mostly be used in output presentation.

I am going to just pass it on to the feeder_stats.json file as a dict